Fix native linux support without wine

snaggen opened the pull request:
This pull request removes the wine dependency for linux. I also created an abstract sdk helper class, and then one for each os to keep all the os specific things in one place. It felt cleaner than having it in the rest of the code. Also, I only use Linux, so my changes is not tested with anything else, but I tried to keep the commands for windows and mac... so it SHOULD work... which most probably might indicate that it have a chance of working.... 
Anyway, on Linux I'm able to use the run button to test my watchface in a simulator with no wine installed. 

However, the Garmin Linux SDK is using VERY old deps
so, to get that to work you might need to find and setup
the dependencies manually. For Fedora 27/28 I setup the following libs:
libjavascriptcoregtk-1.0.so -> libjavascriptcoregtk-1.0.so.0.16.19
libjavascriptcoregtk-1.0.so.0 -> libjavascriptcoregtk-1.0.so.0.16.19
libjavascriptcoregtk-1.0.so.0.16.19
libjpeg.so.8 -> libjpeg.so.8.0.2
libjpeg.so.8.0.2
libwebkitgtk-1.0.so -> libwebkitgtk-1.0.so.0.22.17
libwebkitgtk-1.0.so.0 -> libwebkitgtk-1.0.so.0.22.17
libwebkitgtk-1.0.so.0.22.17

Fixes #12

snaggen committed:
Fix native linux support without wine

However, the Garmin Linux SDK is using VERY old deps
so, to get that to work you might need to find and setup
the dependencies manually. For Fedora 27/28 I setup the following libs:
libjavascriptcoregtk-1.0.so -> libjavascriptcoregtk-1.0.so.0.16.19
libjavascriptcoregtk-1.0.so.0 -> libjavascriptcoregtk-1.0.so.0.16.19
libjavascriptcoregtk-1.0.so.0.16.19
libjpeg.so.8 -> libjpeg.so.8.0.2
libjpeg.so.8.0.2
libwebkitgtk-1.0.so -> libwebkitgtk-1.0.so.0.22.17
libwebkitgtk-1.0.so.0 -> libwebkitgtk-1.0.so.0.22.17
libwebkitgtk-1.0.so.0.22.17

liias commented:
Yay, first pull request. Thanks! Will review this or next week.

snaggen commented:
Realized I forgot to clean it up, will do that tomorrow I think

liias commented on src/main/java/io/github/liias/monkey/project/sdk/SdkHelper.java:
```diff
@@ -0,0 +1,32 @@
+	protected final Sdk sdk;
+
+	SdkHelper(Sdk sdk) {
+		this.sdk = sdk;
```
As sdk is unused here I think this helper class could be simpler if having only static methods and no subclasses.

snaggen commented:
So, I reworked the SdkHelper to remove the subclasses, and allow static access. As you said, it made things much cleaner. 
I squashed the cleanup commit, to avoid commit log pollution.

liias commented:
Alright, thanks!
I tested compiling and running in simulator on Windows.

Created rel 0.3.8. JetBrains has manual approval process, so expect it in a day or two.
